eyetracking/bids_scripts/friends_driftCor.py

```diff
@@ -0,0 +1,551 @@
+    '''
+    frame_dur = 1/fps
+    gaze_perFrame = np.empty(frame_count, dtype='object')
+
+    frame_vals = []
+    i = 0
+
+    for frame_num in range(frame_count):
+        while i < len(time_vals) and time_vals[i] < (frame_num + 1) * frame_dur:
+            frame_vals.append([time_vals[i], x_vals[i], y_vals[i]])
+            i += 1
+        gaze_perFrame[frame_num] = frame_vals
+        frame_vals = []
+
+    return gaze_perFrame
+
+
+def get_distances(
+    frame_count: int,
+    gazes_perframe: np.array,
+    DGvals_perframe: np.array,
+    use_deepgaze: bool=False,
+    x_mass: float=0.5,
+    y_mass: float=0.605,
+) -> tuple:
+    '''
+    Calculates distances between mean gaze (averaged per frame) and DeepGaze
+    prediction.
+    If use_deepgaze is False, then calculates distances to target centers
+    of mass x = 0.5 and y = 0.605
+    '''
+    frame_nums = []
+
+    et_x = []
+    et_y = []
+
+    dist_x = []
+    dist_y = []
+
+    dg_x = []
+    dg_y = []
+
+    for i in range(frame_count):
+        if len(gazes_perframe[i]) > 0:
+            x_et, y_et = np.mean(np.array(gazes_perframe[i])[:, 1:], axis=0)
+
+            if use_deepgaze:
+                """
+                ONLY use only "high confidence" Deepgaze points for frames in
+                which a single local maximum was identified
+                (>.8 max, >20 pixels of distance).
+                """
+                if len(DGvals_perframe[i]) == 1:
+                    x_dg, y_dg = DGvals_perframe[i][0][1:]
+
+                    dist_x.append(x_et - x_dg)
+                    dist_y.append(y_et - y_dg)
+
+                    et_x.append(x_et)
+                    et_y.append(y_et)
+
+                    dg_x.append(x_dg)
+                    dg_y.append(y_dg)
+
+                    frame_nums.append(i)
+
+            else:
+                et_x.append(x_et)
+                et_y.append(y_et)
+
+                dist_x.append(x_et - x_mass)
+                dist_y.append(y_et - y_mass)
+
+                frame_nums.append(i)
+
+    return (et_x, et_y), (dist_x, dist_y), (dg_x, dg_y), frame_nums
+
+
+def median_clean(frame_times, et_x, et_y, dist_x, dist_y):
+    '''
+    Within bins of 1/100 the number of frames,
+    select only frames where distance between gaze and deepgaze falls within 0.6 stdev of the median
+    These frames most likely reflect when deepgaze and gaze "look" at the same thing
+    '''
+    jump = int(len(dist_x)/100)
+    idx = 0
+    gap = 0.6 # interval of distances included around median, in stdev
+
+    filtered_times = []
+    filtered_x = []
+    filtered_y = []
+    filtered_distx = []
+    filtered_disty = []
+
+    current_medx = np.median(np.array(dist_x)[idx:idx+jump])
+    current_medy = np.median(np.array(dist_y)[idx:idx+jump])
+    stdevx = np.std(np.array(dist_x)[idx:idx+jump])
+    stdevy = np.std(np.array(dist_y)[idx:idx+jump])
```

Comment:
why computing that first out of the for loop under, where it is computed again?

Reply:
because I screwed up my function. see below...